manifests/config.pp

```diff
@@ -1,22 +1,24 @@
 # == Class: marathon::config
 #
 class marathon::config(
-  $conf_dir_base = '/etc/marathon',
-  $conf_dir_name = 'conf',
-  $owner         = 'root',
-  $group         = 'root',
-  $master        = undef,
-  $zookeeper     = undef,
-  $options       = {},
-  $env_var       = {},
-  $manage_logger = true,
-  $logger        = 'logback',
-  $log_dir       = '/var/log/marathon',
-  $log_filename  = 'marathon.log',
-  $log_level     = 'info',
-  $ulimit        = undef,
-  $java_home     = undef,
-  $java_opts     = '-Xmx512m',
+  $conf_dir_base                    = '/etc/marathon',
+  $conf_dir_name                    = 'conf',
+  $owner                            = 'root',
+  $group                            = 'root',
+  $master                           = undef,
+  $zookeeper                        = undef,
+  $options                          = {},
+  $env_var                          = {},
+  $manage_logger                    = true,
+  $logger                           = 'logback',
+  $log_dir                          = '/var/log/marathon',
+  $log_filename                     = 'marathon.log',
+  $log_level                        = 'info',
+  $ulimit                           = undef,
+  $secret                           = undef,
+  $mesos_authentication_secret_file = '/etc/marathon/.secret',
```

Comment:
Ok.. I've thought about this a bit and can we maybe do:
```puppet
$mesos_auth_principal   = undef,
$mesos_auth_secret      = undef,
$mesos_auth_secret_file = '/etc/marathon/.secret',
```

And then:
```puppet
if ($mesos_auth_principal != undef) and ($mesos_auth_secret != undef) {
  $secret_options = {
    'mesos_authentication_principal'   => $mesos_auth_principal,
    'mesos_authentication_secret_file' => $mesos_auth_secret_file,
  }
} else { ...
```

Sorry for chopping and changing... `$mesos_authentication_secret_file` seems a bit long and also the secret is not much good without a principal.

Does this API sound good to you?